### PR TITLE
Simple check for presence of HA-based configuration

### PR DESCRIPTION
**Pull request by jsuchome**
This is to be used by crowbar pre-checks, run before the upgrade

### CODE REVIEW

File: crowbar_framework/app/models/api/pacemaker.rb [hunk 1]
```diff
@@ -17,6 +17,20 @@
 module Api
   class Pacemaker < Tableless
     class << self
+      # Check for presence of HA setup, which is a requirement for non-disruptive upgrade
+      def ha_presence_check
+        ret = {}
```
Comment (MaximilianMeister) [inline crowbar_framework/app/models/api/pacemaker.rb]: matter of taste but I think we could use tap here, which always returns the hash as we want it.
and currently you are overriding the array if both errors happen, I would suggest sth like:

```ruby
{}.tap do |ret|
  unless repocheck["ha"]["available"]
    ret[:errors] = []
    ret[:errors].push(I18n.t("api.pacemaker.ha_not_installed"))
    return ret
  end
  founders = NodeObject.find("pacemaker_founder:true AND pacemaker_config_environment:*")
  if founders.empty?
    ret[:errors] ||= []
    ret[:errors].push(I18n.t("api.pacemaker.ha_not_configured"))
  end
end
```

Comment (aspiers) [inline crowbar_framework/app/models/api/pacemaker.rb]: That's strangely verbose.  Why not just this?

```ruby
unless repocheck["ha"]["available"]
  return { errors: [ I18n.t("api.pacemaker.ha_not_installed") ] }
end
founders = NodeObject.find("pacemaker_founder:true AND pacemaker_config_environment:*")
founders.empty? ? { errors: [ I18n.t("api.pacemaker.ha_not_configured") ] } : {}
```

Comment (jsuchome) [inline crowbar_framework/app/models/api/pacemaker.rb]: Does this really matter that much?

**Comment by jsuchome** [comment]
I do not have any special reason for that `tap`, I've added it based on @MaximilianMeister suggestion. In general, I do not find it better or worse with my original approach, Max's tap, or Adam's shorter one.  A little bit more or less lines, and all examples are readable ... then I just tend to adapt to the needs of reviewers to get the stuff done.

**Comment by aspiers** [comment]
I'm not religiously against the use of `tap` but I would just like to understand if there is a reason for using it which I'm missing.

**Comment by jsuchome** [comment]
> but I would just like to understand if there is a reason for using it which I'm missing.

I don't know, I believe it's matter of taste. As I wrote, I followed Max's recommendation. @MaximilianMeister ?

**Comment by MaximilianMeister** [comment]
> I don't know, I believe it's matter of taste. As I wrote, I followed Max's recommendation. @MaximilianMeister ?

I don't mind either way is fine with me

> I'm not religiously against the use of tap but I would just like to understand if there is a reason for using it which I'm missing.

it was just a suggestion of how I would do it, doesn't mean that there is no better way of doing it

**Comment by jsuchome** [comment]
Adapted according to @aspiers suggestions

**Comment by MaximilianMeister** [comment]
gating error is unrelated, and this code is not used at all. we need it as well for backporting it to be used by the prechecks API

**Pull request merged**